feat: call known lineage variants (#392)
IKIM-Essen/uncovar: docs/consumer-guide/report-structure.md

# Understanding the Report

UnCoVar automatically generates a HTML-based report for a detailed insight into
the analysed patient or environmental samples

## Section 1: Overview

### Analysis overview table (overview.html) for all entered samples

- **Sample name**
- **Species contamination** (especially interesting for samples not prepared
  with amplicon tiled preparation)
- Number of entered **raw reads** and number of reads **after species
  filtering** and **primer clipping**
- Length of the **largest contig** after [_de novo_ assembly](https://en.wikipedia.org/wiki/De_novo_sequence_assemblers)
- Length of the **_de novo_ sequence** after reference guided scaffolding with
  [RaGOO](https://github.com/malonge/RaGOO)
- Length of the **consensus/pseudo sequences** generated by including
  called variants with [Varlociraptor](https://varlociraptor.github.io) into
  the Wuhan-Hu-1 Reference
  [`NC_045512.2`](https://www.ncbi.nlm.nih.gov/nuccore/1798174254)

> - **Called lineage** from [Pangolin](https://github.com/cov-lineages/pangolin)

Reviewer: ```suggestion
- Pango Lineage: Called lineage in [Pango nomenclature](https://cov-lineages.org/).
- WHO Label: Lineage name determined by the [WHO](https://www.who.int/en/activities/tracking-SARS-CoV-2-variants/).
```

> - **[SNVs](https://en.wikipedia.org/wiki/Single-nucleotide_polymorphism)**

Reviewer: ```suggestion
- VOC Mutations: Mutations that occur in Variants of Concern (VOC). These VOCs typically have: Increase in transmissibility or detrimental change in COVID-19 epidemiology; OR Increase in virulence or change in clinical disease presentation; OR Decrease in effectiveness of public health and social measures or available diagnostics, vaccines, therapeutics.
Other Mutations: Other Mutations that occur in the sample.
```

  of specific concern
  ([VOC](https://en.wikipedia.org/wiki/Variant_of_concern)) and other found variants

## Section 2: Variant Call Details

> ### Analysis of present VOC mutations and top 5 possible lineages per sample

Reviewer: ```suggestion
### VOC Similarity

Comparison of found mutations in the sample with mutations of Variants of Concern. Show the top 5 Variants of Concern to the sample in terms of similarity.
```

> - **Mutations** found

Reviewer: ```suggestion
- Mutations: All mutations found in the sample.
```

> - **Probability** and **Frequency** both calculated via a-posteriori probability

Reviewer: ```suggestion
- Probability: A-posteriori probability that the mutations are present.
- Frequency: Variant allele frequency (VAF). The percentage of sequence reads observed matching a specific DNA variant divided by the overall coverage at that position.
```

  and observed variant allele frequency (VAF) of each mutation

> - **Top 5 lineages** based on feature similarity per top lineage, calculated with

Reviewer: ```suggestion
- Top 5 lineages: The most similar variants of concerns to the sample The [Jaccard index](https://en.wikipedia.org/wiki/Jaccard_index) is used as a measure of similarity.
```
